src/util/piped_process.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CPROVER_UTIL_PIPED_PROCESS_H
+#define CPROVER_UTIL_PIPED_PROCESS_H
+
+// TODO: Revisit buffer size and storage location.
+// Note that this may impact number of characters correctly communicated
+// by the pipe, to check/fix.
+#define BUFSIZE 2048
+
+class piped_processt
+{
+  enum class process_statet
+  {
+    NOT_CREATED,
+    CREATED,
+    STOPPED
+  };
+
+  char **split_command_args(const std::string &command);
+
+protected:
+  FILE *response_stream;
+  FILE *command_stream;
+  int pipe_input[2];
+  int pipe_output[2];
+  process_statet process_state;
+
+public:
+  bool send(const std::string &message);
```

Comment:
In some implementations of the STL it used to be possible to create an `ostream` or an `istream` based on a file descriptor.  If that were possible it would give a much cleaner and more consistent interface to the pipe.

Reply:
https://gist.github.com/rajatkhanduja/2012695 gives an example of this.